libmscore/chord.cpp

```diff
@@ -1448,21 +1419,45 @@ qreal Chord::defaultStemLength() const
 
 //---------------------------------------------------------
 //   minAbsStemLength
+//    get minimum stem length with tremolo
 //---------------------------------------------------------
 
 qreal Chord::minAbsStemLength() const
       {
-      if (!_tremolo || _tremolo->twoNotes() || !_tremolo->placeMidStem())
+      const qreal sw = score()->styleS(Sid::tremoloStrokeWidth).val();
+      const qreal td = score()->styleS(Sid::tremoloDistance).val();
+      int beamLvl = beams();
+      const qreal beamDist = beam() ? beam()->beamDist() : (sw * spatium());
+
+      if (!_tremolo)
             return 0.0;
 
-      int beamLvl = beams();
-      const bool hasHook = (beamLvl > 0) && !beam();
-      if (hasHook)
-            ++beamLvl; // reserve more space for stem with both hook and tremolo
-      const qreal beamDist = beam() ? beam()->beamDist() : (0.5 * spatium());
-      const qreal tremoloSpacing = 0.5 * spatium(); // TODO: style setting
+      // single-note tremolo
+      else if (!_tremolo->twoNotes()) {
+            _tremolo->layout(); // guarantee right "height value"
 
-      return beamLvl * beamDist + _tremolo->height() + 2 * tremoloSpacing;
+            qreal height;
+            if (up())
+                  height = downPos() - _tremolo->pos().y();
+            else
+                  height = _tremolo->pos().y() + _tremolo->height() - upPos();
+            const bool hasHook = beamLvl && !beam();
+            if (hasHook)
+                  beamLvl += (up() ? 4.0 : 2.5); // reserve more space for stem with both hook and tremolo
```

Comment:
Here it seems a mistake. beamLvl is an int type, and fractional ones are added

Reply:
Will change to `qreal`.

Reply:
Nope, `2` is the best choice, and in this way it is the same as what I'd seen before, since `2.5` is automatically converted to `2` then.